src/silly_sprites/opengl/shader_program.cpp

```diff
@@ -0,0 +1,141 @@
+#include "shader_program.hpp"
+#include "utils.hpp"
+
+namespace sly::gl {
+    static constexpr char const* fallback_vertex_source{ R"(
+            #version 330 core
+            layout (location = 0) in vec3 aPos;
+
+            void main()
+            {
+                gl_Position = vec4(aPos.x, aPos.y, aPos.z, 1.0);
+            }
+        )" };
+    static constexpr char const* fallback_fragment_source{ R"(
+            #version 330 core
+            out vec4 FragColor;
+            
+            void main()
+            {
+                FragColor = vec4(255.0f, 0.0f, 255.0f, 1.0f);
+            }    
+        )" };
+
+    [[nodiscard]] constexpr std::string_view ShaderProgram::get_name_from_type(Type const type) {
+        switch (type) {
+            case Type::Vertex:
+                return "VERTEX";
+            case Type::Geometry:
+                return "GEOMETRY";
+            case Type::Fragment:
+                return "FRAGMENT";
+            default:
+                return "INVALID";
+        }
```

Comment:
I think this would be better:
```suggestion
        switch (type) {
            case Type::Vertex:
                return "VERTEX";
            case Type::Geometry:
                return "GEOMETRY";
            case Type::Fragment:
                return "FRAGMENT";
        }
        return "INVALID";
```
Reasoning: The compiler should warn if you forget any of the enum variants inside the `switch` statement. But if you have a `default` label, the compiler will never warn you.

Reply:
good idea.
i changed that but MSVC does not warn.
maybe that is a MSVC thing or we have to enable some warnings in CMAKE.

Reply:
Maybe that's a problem with MSVC then, but GCC and Clang should warn in that case.